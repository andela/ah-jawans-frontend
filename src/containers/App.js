/* eslint-disable import/no-named-as-default */
import '@babel/polyfill';
import React from 'react';
import { Route, BrowserRouter, Switch } from 'react-router-dom';
import Signup from './signupContainer';
import Login from './loginView';
import Profile from '../components/Profile';
import '../assets/css/style.scss';
import '../assets/scss/main.scss';
import Home from '../components/Home';
import PageNotFound from '../components/PageNotFound';
import PasswordReset from './passwordReset';
import UpdatePassword from './updatePassword';


const App = () => (
  <BrowserRouter>
    <Switch>
      <Route exact path="/" component={Home} />
      <Route path="/signup" component={Signup} />
      <Route path="/login" component={Login} />
      <Route exact path="/profile" component={Profile} />
      <Route path="/passwordReset" component={PasswordReset} />
      <Route path="/updatePassword" component={UpdatePassword} />
      <Route component={PageNotFound} />
    </Switch>
  </BrowserRouter>
);

export default App;
