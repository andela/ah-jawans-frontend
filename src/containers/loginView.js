import React, { Component } from 'react';
import { connect } from 'react-redux';
import queryString from 'query-string';
import PropTypes from 'prop-types';
import LoginComponet from '../components/Login';
import '../assets/scss/components/login.scss';
import loginUserAction from '../redux/actions/loginAction';
import Error from '../components/common/errors';
import socialLoginAction from '../redux/actions/sosialLoginAction';
import { postDataThunk } from '../redux/thunks';
import computerHand from '../assets/images/computerHand.jpg';

export class Login extends Component {
  state = {
    user: {
      email: '',
      password: '',
    },
  };

  componentDidMount() {
    const { token } = queryString.parse(this.props.location.search);
    socialLoginAction({ token });
  }

  handleChange = (e) => {
    const user = { ...this.state.user };
    user[e.target.name] = e.target.value;
    this.setState({ user });
  };

  handleSubmit = async (e) => {
    e.preventDefault();
    const { user } = this.state;
    await this.props.postDataThunk('post', 'users/login', loginUserAction, user);
    if (!this.props.userCredentials.errors) {
      const { userCredentials: { data: { username, token } } } = this.props;
      localStorage.setItem('username', username);
      localStorage.setItem('token', token);
      this.props.history.push('/profile');
    }
  };

  render() {
    return (
      <div className="container">
        {(this.props.userCredentials.errors) && <Error
          errors={this.props.userCredentials.errors}
        />}
        <div className="row">
          <div className="col loginLeftSide">
            <h1>Login</h1>
            <h2>Authors Haven</h2>
            <img className='images' src={computerHand}/>
          </div>
          <div className="col-md-6 myForm">
            <h4> Signin </h4>
            <LoginComponet
              onChange={this.handleChange}
              onSubmit={this.handleSubmit}
              user={this.state.user}
            />
          </div>
        </div>
      </div>
    );
  }
}

Login.propTypes = {
  postDataThunk: PropTypes.func,
  userCredentials: PropTypes.object,
  history: PropTypes.object,
  location: PropTypes.object,
  socialLogin: PropTypes.func,
};

const mapStateToProps = (state) => ({
  userCredentials: state.userCredentials.userCredentials,
  errors: state.errors,
});

const actionCreator = {
  postDataThunk, socialLoginAction,
};

export default connect(
  mapStateToProps,
  actionCreator,
)(Login);
