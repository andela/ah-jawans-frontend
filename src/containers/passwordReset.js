/* eslint-disable no-unused-expressions */
/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Card from '../components/Card';
import Modal from '../components/Modal';
import { requestPasswordRequest } from '../redux/actions/passwordReset';
import '../assets/scss/components/__passwordReset.scss';
import { postDataThunk } from '../redux/thunks';

export class PasswordReset extends Component {
  state = {
    email: '',
    modal: false,
    UserNotFoundModal: false,
    errors: '',
  };

  componentWillReceiveProps(nextProps) {
    const { resetPassword } = nextProps;
    resetPassword.reset && !resetPassword.reset.errors
      ? this.setState({ modal: true, errors: resetPassword.reset.errors })
      : this.setState({ ...this.state });
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    await this.props.postDataThunk(
      'post',
      'users/passwordreset',
      requestPasswordRequest,
      this.state,
    );
  };

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value, emailError: '' });
  };

  render() {
    const { emailError, modal } = this.state;
    if (modal) {
      return <Modal />;
    }
    return (
      <Card className='card'>
        <Link to='/' style={{ margin: '0 auto' }} />
        <h1 className='card__title'>Authors Haven</h1>
        <br />
        <h2 className='card__heading'>Reset Password</h2>
        <br />
        <div className='card__errors'>{emailError}</div>
        <h3 className='card__label'>Email</h3>
        <form className='card__form' onSubmit={this.handleSubmit}>
          <input
            className='card__input'
            type='email'
            id='email'
            name='email'
            onChange={this.handleChange}
            placeholder='Enter your email'
          />
          {this.props.resetPassword.reset && this.props.resetPassword.reset.errors ? (
            <span className='text-danger'>{this.props.resetPassword.reset.errors}</span>
          ) : ''}
          <br />
          <button className='card__button' type='submit' name='submit'>
            {' '}
            Reset
          </button>
        </form>
      </Card>
    );
  }
}

PasswordReset.propTypes = {
  reset: PropTypes.func,
  history: PropTypes.instanceOf(Object).isRequired,
  requestToReset: PropTypes.func,
};

const mapStateToProps = (state) => ({
  errors: state.resetPassword,
  resetPassword: state.resetPassword,
});

const actionCreator = {
  postDataThunk,
};

export default connect(
  mapStateToProps,
  actionCreator,
)(PasswordReset);
