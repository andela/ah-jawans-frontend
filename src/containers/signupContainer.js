/* eslint-disable import/no-named-as-default */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import queryString from 'query-string';
import PropTypes from 'prop-types';
import Error from '../components/common/errors';
import signupUserAction from '../redux/actions/signup.actions';
import { postDataThunk } from '../redux/thunks/index';
import SignupComponent from '../components/pages/Signup';
import socialLoginAction from '../redux/actions/sosialLoginAction';
import computerHand from '../assets/images/computerHand.jpg';

export class Signup extends Component {
  componentDidMount() {
    const { token } = queryString.parse(this.props.location.search);
    socialLoginAction({ token });
  }

  state = {
    user: {
      username: '',
      email: '',
      password: '',
    },
  };

  handleInputChange = (e) => {
    const user = { ...this.state.user };
    user[e.target.name] = e.target.value;
    this.setState({ user });
  };

  handleSubmit = async (e) => {
    e.preventDefault();
    const { user } = this.state;
    await this.props.postDataThunk('post', '/users', signupUserAction, user);
    if (!this.props.signupSuccess.errors) this.props.history.push('/login');
  };

  render() {
    return (
      <div className="container">
        {this.props.signupSuccess.errors && <Error
          errors={this.props.signupSuccess.errors}
        />}
        <div className="row">
          <div className="col-7 container__leftSide">
            <h1>Join us</h1>
            <h5>Authors Haven</h5>
            <img
              className="container__images"
              src={computerHand}
            />
          </div>
          <div className="col-5 container__rightSide">
            <h4> Sign Up </h4>
            <SignupComponent
              onChange={this.handleInputChange}
              onSubmit={this.handleSubmit}
              user={this.state.user}
            />
          </div>
        </div>
      </div>
    );
  }
}
Signup.propTypes = {
  location: PropTypes.object,
  socialLogin: PropTypes.func,
  signupSuccess: PropTypes.object,
  postDataThunk: PropTypes.func,
  history: PropTypes.object,

};

const mapStateToProps = (state) => ({
  signupSuccess: state.signupSuccess.signupSuccess,
  errors: state.errors,
});

const actionCreator = {
  postDataThunk, socialLoginAction,
};

export default connect(
  mapStateToProps,
  actionCreator,
)(Signup);
