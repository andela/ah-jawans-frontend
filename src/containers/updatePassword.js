/* eslint-disable no-alert */
import React, { Component } from 'react';
import queryString from 'query-string';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Card from '../components/Card';
import { changePassword } from '../redux/actions/passwordReset';
import { postDataThunk } from '../redux/thunks/index';

export class UpdatePassword extends Component {
  state = {
    passwordError: '',
    confirmationError: '',
    password1: '',
    password2: '',
    token: null,
  };

  componentDidMount() {
    const { token } = queryString.parse(this.props.location.search);
    this.setState({ token });
  }

  // eslint-disable-next-line class-methods-use-this
  componentWillReceiveProps(nextProps) {
    if (nextProps.resetPassword.passwordUpdate) {
      window.location.href = '/login';
    }
  }

  handleChange = (e) => {
    this.setState({
      [e.target.id]: e.target.value,
      passwordError: '',
      confirmationError: '',
    });
  };

  changePass = async (e) => {
    const { password1, password2 } = this.state;
    e.preventDefault();
    if (password1 !== password2) {
      window.alert('Passwords do not match', 'danger');
    } else {
      const password = this.state.password1;
      await this.props.postDataThunk(
        'post',
        `users/passwordreset/${this.state.token}`,
        changePassword,
        { password },
      );
    }
  };

  render() {
    const { passwordError, confirmationError } = this.state;
    return (
      <Card className='card'>
        <Link to='/' style={{ margin: '0 auto' }} />
        <h1 className='card__title'>Authors Haven</h1>
        <br />
        <h2 className='card__heading'>Update Password</h2>
        <form onSubmit={this.changePass}>
          <br />
          <h3 className='card__label'>Password</h3>
          <input
            className='card__input'
            type='password'
            onChange={this.handleChange}
            placeholder='Enter your password'
            id='password1'
          />
          <div className='card__passError'>{passwordError}</div>
          <br />
          <h3 className='card__label'>Confirm Password</h3>
          <input
            className='card__input'
            type='password'
            onChange={this.handleChange}
            placeholder='Enter password confirmation'
            id='password2'
          />
          <div className='card__confError'>{confirmationError}</div>
          <br />
          <button className='card__button' type='submit' name='submit'>
            Update
          </button>
        </form>
      </Card>
    );
  }
}

UpdatePassword.propTypes = {
  match: PropTypes.instanceOf(Object).isRequired,
  changePassword: PropTypes.func,
  history: PropTypes.instanceOf(Object).isRequired,
  location: PropTypes.instanceOf(Object).isRequired,
  resetPassword: PropTypes.func.isRequired,
  passwordUpdate: PropTypes.func.isRequired,
  setAlert: PropTypes.func.isRequired,
  postDataThunk: PropTypes.func,
};

const mapStateToProps = (state) => ({
  updatedPassword: state.update,
  resetPassword: state.resetPassword,
});

const actionCreator = {
  postDataThunk,
};

export default connect(
  mapStateToProps,
  actionCreator,
)(UpdatePassword);
