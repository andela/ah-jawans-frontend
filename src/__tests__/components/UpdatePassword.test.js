import React from 'react';
import { shallow } from '../../../config/enzymeConfig';
import { UpdatePassword } from '../../containers/updatePassword';


const props = {
  resetPassword: jest.fn(),
  history: {
    push: jest.fn(),
  },
  match: {
    params: {
      token: 'We have a token here',
    },
  },
  location: { search: 'url' },
  postDataThunk: () => Promise.resolve({}),
};
const component = shallow(<UpdatePassword {...props} />);
describe('Update password', () => {
  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });
  it('should test redirection to login page', () => {
    component.instance().componentWillReceiveProps({ resetPassword: { passwordUpdate: 'passwod' } });
  });
  it('should test handleChange function', () => {
    component.instance().handleChange({ target: { id: 2, value: 'value' } });
  });
  it('should test Change pass function', () => {
    window.alert = jest.fn();
    component.instance().changePass({ preventDefault: jest.fn() });
  });
});
