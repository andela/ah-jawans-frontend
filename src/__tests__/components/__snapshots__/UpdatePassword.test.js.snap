// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update password should render without crashing 1`] = `ShallowWrapper {}`;
